pkgs/development/python-modules/opensfm/default.nix

```diff
@@ -0,0 +1,110 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cmake
+, opencv
+, opencv4
+, ceres-solver
+, suitesparse
+, metis
+, eigen
+, pkg-config
+, pybind11
+, numpy
+, pyyaml
+, lapack
+, gtest
+, gflags
+, glog
+, pytestCheckHook
+, networkx
+, pillow
+, exifread
+, gpxpy
+, pyproj
+, python-dateutil
+, joblib
+, repoze_lru
+, xmltodict
+, cloudpickle
+, scipy
+, sphinx
+, matplotlib
+, fpdf
+}:
+
+buildPythonPackage rec {
+  pname = "OpenSfM";
+  version = "0.5.2";
+  src = fetchFromGitHub {
+    owner = "mapillary";
+    repo = "OpenSfM";
+    # Fixing by commit-id until the upstream issues are solved
+    rev = "2a1c9be07f47f6e26f87fedf8275fcf8ff7b8487";
+    sha256 = "sha256-CARPFuYypgV1YcqZp//Qws8YrV25Z+w/dWB7MbFAKys=";
+    fetchSubmodules = true;
+  };
+  nativeBuildInputs = [ cmake pkg-config sphinx ];
+  buildInputs = [
+    opencv
+    ceres-solver.dev
+    suitesparse
+    metis
+    eigen
+    lapack
+    gflags
+    gtest
+    glog
+  ];
+  propagatedBuildInputs = [
+    numpy
+    scipy
+    pyyaml
+    opencv4
+    networkx
+    pillow
+    matplotlib
+    fpdf
+    exifread
+    gpxpy
+    pyproj
+    python-dateutil
+    joblib
+    repoze_lru
+    xmltodict
+    cloudpickle
+  ];
+  checkInputs = [ pytestCheckHook ];
+  dontUseCmakeBuildDir = true;
+  cmakeFlags = [
+    "-Bcmake_build"
+    "-Sopensfm/src"
+  ];
+  patches = [
+    ./fix-cmake.patch
+    ./fix-scripts.patch
+    ./fix-report-generation.patch
```

Comment:
It would be great if we could try to upstream the none NixOS specific changes.

Reply:
You're right, I should get `fix-report-generation.patch` merged upstream before this PR can get a go (I've opened a quick [PR](https://github.com/mapillary/OpenSfM/pull/854)), but the other two
- make distribution-related adjustments (like fixing shebangs and pointing cmake at different places)
- are probably bit harder to get approved upstream

Reply:
But in their CI they seem to have passing tests, it's quite likely nixpkgs just packages a different release of `fpdf` 🤔

Reply:
You can also fetch the patch from the PR. Than everyone knows it is already being upstreamed.

We have the latest and greated fpdf version https://repology.org/project/python:fpdf/versions

Reply:
> You can also fetch the patch from the PR. 

Sounds reasonable. Done

> We have the latest

I thought to check if they use an old one, but in fact they use an alternative package, `fpdf2` =\ Maybe will add it tomorrow